sql/utils/workflow_audit.py

```diff
@@ -300,6 +300,22 @@ def create_audit(self) -> str:
             create_user=create_user,
             create_user_display=create_user_display,
         )
+        if (
+            audit_setting.auto_pass
+            and self.workflow.status == "workflow_autoreviewwrong"
+        ):
```

Comment:
无需判断是否 auto_pass, 逻辑更清晰

```suggestion
        if self.workflow.status == "workflow_autoreviewwrong":
```

Reply:
我的想法是手工确认的模式下，不执行自动驳回

Reply:
不行, 你的这个需求是 breaking change 了, 需要做系统配置控制行为

Reply:
如果你需要, 可以独立成一个函数, `should_reject` , 然后你定制开发时覆盖这个函数即可. 这样也可以控制审批流的行为.

参考:
* https://github.com/hhyo/Archery/blob/52ce7595d4b3b6e294b49c4d9a4ce10a0992d60f/archery/settings.py#L71C5-L71C20 
* https://github.com/hhyo/Archery/blob/52ce7595d4b3b6e294b49c4d9a4ce10a0992d60f/sql/utils/workflow_audit.py#L746

Reply:
意思是对于错误工单，提交后可以修改？
这样当然是好的，但是这个状态怎么设置，没搞清楚

Reply:
错误工单提交的情况下, 提交者可以拿着驳回的工单跟系统管理员沟通, 不用使用截图之类的方法就可以顺畅的沟通.

你说的状态设置具体指的是什么? 我没看明白

Reply:
没搞懂“breaking change”是什么意思。
错误工单不应该驳回，还要参数控制？

Reply:
> 我的想法是手工确认的模式下，不执行自动驳回

错误工单应该驳回, 但是你刚刚的观点是, **只有自动审核通过, 且需要驳回的时候才驳回**, 我的观点是, **需要驳回就应该驳回, 不需其他条件**

如果说你想做 **只有自动审核通过, 且需要驳回的时候才驳回** , 那这个行为和之前的版本不一致, 是需要参数控制的.

Reply:
我表达的意思是在自动审核模式下，需要驳回的就驳回，而不是自动审核通过；在人工确认模式下，尽管有需要驳回的，仍可通过。
is_auto_review函数表达的仅仅是是否符合自动审核的条件，generate_audit_setting函数第一行，表达的就是只要符合自动审核条件，就不管其它了（最终体现在create_audit），我要修改的是这个逻辑

Reply:
我表达的意思是在自动审核模式下，需要驳回的就驳回，而不是自动审核通过；在人工确认模式下，尽管有需要驳回的，仍可通过。
is_auto_review函数表达的仅仅是是否符合自动审核的条件，generate_audit_setting函数第一行，表达的就是只要符合自动审核条件，就不管其它了（最终体现在create_audit），我要修改的是这个逻辑

Reply:
好的，那就需要参数控制，辛苦实现下。

Reply:
好的，那还是一样，你的新逻辑带来了breaking change，需要系统配置项来支持自定义，或者你也可以自己二次开发实现这个功能。

Reply:
前面我也问了，这个“breaking change”是什么意思，没搞明白。
在自动审核模式下，存在错误的工单，不应该驳回吗？ 还需要参数进行开关？

Reply:
breaking change 就是和之前版本不一致的行为，对于你的两个问题我就不重复回答了，请参照上方粗体部分的说明。

Reply:
我觉得需要分析的是，是否存在其它场景，不符合我实现的逻辑。

Reply:
这里可能还是有一些矛盾在的哈，我觉得我观点都表达到位了。如果没有代码改变，新的观点或论据，我就不反复沟通了。

Reply:
目前的实现与原来的逻辑基本没有差别了，只是在判断是否自动通过时，增加了对存在审核错误的检查，其实和最早的实现类似。
另外增加了一个系统参数，是否要允许错误脚本提交（仅自动审核模式下），目前是在客户端脚本里进行判断。